stdgo/reflect/Reflect.hx

```diff
@@ -943,12 +1133,65 @@ class Value {
 			case GT_string:
 				return (val : GoString);
 			case _:
-				throw "Value.String not implemented for " + underlyingType.string();
+				throw new ValueError("String", underlyingType.kind());
 		}
 		return "";
 	}
 }
 
+// A ValueError occurs when a Value method is invoked on
+// a Value that does not support it. Such cases are documented
+// in the description of each method.
+@:rtti class ValueError extends StdGo2hx {
+	var method:GoString;
+	var kind:Kind;
+
+	public inline function new(m:GoString, k:Kind) {
+		super(Type.getClassName(ValueError));
+		method = m;
+		kind = k;
+	}
+
+	public function __copy__()
+		return new ValueError(method, kind);
+
+	@:to public function __promote()
+		return new AnyInterface({value: this, typeName: _typeName_});
+
+	public function toString() {
+		return (this.error() : String);
+	}
+
+	public function error():GoString {
+		if (this.kind == invalid) {
+			return "reflect: call of " + this.method + " on zero Value";
+		}
+		return "reflect: call of " + this.method + " on " + this.kind.string() + " Value";
+	}
+}
+
+// a local generic error
+@:rtti class Error extends StdGo2hx {
```

Comment:
Needs to implement interface stdgo.StdGoTypes.Error

Reply:
Will do.